### ISSUE TITLE
Selecting custom language profile not working

### ISSUE THREAD

**Issue by lost-echo** [chergert/gnome-builder#44]
Hi,

I'm encountering troubles when trying to assign a different programming language profile to a file in the source code editor. Under "Preferences > Programming Languages" I've configured the profiles "C++" and "C++ Header" to use 4 spaces instead of a tabs and disabled trailing whitespace trimming.

This works fine, but sadly only as long as I use standard filename extensions (*.hh and *.cc in my case) so that Builder automatically chooses a language profile. When creating a *.tcc file for template code, Builder selects the "C" language profile because it doesn't recognize it as a C++ header file.

Selecting "C++ Header" in the popover menu at the top right corner of the source code editor doesn't quite work, as I'm still left with whitespace trimming and tabs instead of spaces.

Any solutions? Thanks!

**Comment by chergert** [comment]
What do you get for "standard::content-type" when running `gvfs-info my-file.tcc`. That is probably what is causing GtkSourceView to think the file is a C file.

Do you have any modelines defined in the file (vim, emacs, kate, etc). Those sometimes mess things up with our file settings layering.

Otherwise, probably just a run of the mill bug we need to fix :)

**Comment by lost-echo** [comment]
The content type is 'text/x-csrc', obviously the one for C sources. Since I don't know what modelines are and I'm using a quite fresh Archlinux installation, I'd guess that isn't the problem ;)

**Comment by chergert** [comment]
Modelines looks something like the following at the top or bottom of your file.

```c
/* ex:set ts=8 noet: */
```

**Comment by lost-echo** [comment]
Ah, no modelines then.. The problem is just that when customizing the language the settings aren't loaded properly (it uses just the defaults):

![screenshot from 2015-11-20 13-45-58](https://cloud.githubusercontent.com/assets/6241153/11300221/786454f2-8f8d-11e5-8be1-67802eea6ac8.png)

**Comment by chergert** [comment]
Did you setup your C++ settings for both headers and source in preferences?

Also note, that we do support editorconfig if you just want to drop a .editorconfig in your source tree.

**Comment by lost-echo** [comment]
Yes, I did configure both profiles, that's why I assumed selecting a custom language in the popover menu should do the trick..

Editorconfig is a very nice thing I didn't know of, thank you! I've created a configuration file at the project root - the desired settings are now loaded when selecting "C++ Header" as language:

```
root = true

[*.{hh,tcc,cc}]
indent_style = space
indent_size = 4
trim_trailing_whitespace = false
insert_final_newline = false
```

![screenshot from 2015-11-20 22-43-37](https://cloud.githubusercontent.com/assets/6241153/11312859/21a0012e-8fd9-11e5-8e54-bc360e66b35e.png)
